taxi/tests/test_views.py

```diff
@@ -0,0 +1,565 @@
+            "password1": "test_password",
+            "password2": "test_password",
+        }
+        self.client.post(url, data)
+        self.assertTrue(
+            Driver.objects.filter(username=data["username"]).exists()
+        )
+
+
+class DriverUpdateViewTests(TestCase):
+    def setUp(self):
+        user = get_user_model().objects.create_user(
+            username="test_user",
+        )
+        self.client.force_login(user)
+        Driver.objects.create(
+            username="test_driver",
+        )
+
+    def test_access_by_url(self):
+        url = "/drivers/2/update/"
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+    def test_access_by_name(self):
+        url = reverse("taxi:driver-update", kwargs={"pk": "2"})
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+    def test_use_correct_template(self):
+        url = reverse("taxi:driver-update", kwargs={"pk": "2"})
+        response = self.client.get(url)
+        self.assertTemplateUsed(response, "taxi/driver_form.html")
+
+    def test_driver_was_updated(self):
+        url = reverse("taxi:driver-update", kwargs={"pk": "2"})
+        data = {
+            "username": "test_driver_updated",
+            "license_number": "ABC12345",
+            "first_name": "test_first_name",
+            "last_name": "test_last_name",
+            "password1": "test_password",
+            "password2": "test_password",
+        }
+        self.client.post(url, data)
+        driver = Driver.objects.get(id=2)
+        driver.refresh_from_db()
+        self.assertEqual(
+            driver.username,
+            data["username"],
+        )
+
+
+class DriverDeleteViewTests(TestCase):
+    def setUp(self):
+        user = get_user_model().objects.create_user(
+            username="test_user",
+        )
+        self.client.force_login(user)
+        Driver.objects.create(
+            username="test_driver",
+        )
+
+    def test_access_by_url(self):
+        url = "/drivers/2/delete/"
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+    def test_access_by_name(self):
+        url = reverse("taxi:driver-delete", kwargs={"pk": "2"})
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+    def test_use_correct_template(self):
+        url = reverse("taxi:driver-delete", kwargs={"pk": "2"})
+        response = self.client.get(url)
+        self.assertTemplateUsed(response, "taxi/driver_confirm_delete.html")
+
+    def test_driver_deleted(self):
+        url = reverse("taxi:driver-delete", kwargs={"pk": "2"})
+        self.assertEqual(Driver.objects.count(), 2)
+        response = self.client.post(url, {"pk": "2"})
+        self.assertEqual(response.status_code, 302)
+        self.assertRedirects(response, reverse("taxi:driver-list"))
+        self.assertTrue(Driver.objects.count(), 1)
```

Comment:
The assertion `self.assertTrue(Driver.objects.count(), 1)` should be `self.assertEqual(Driver.objects.count(), 1)`. `assertTrue` is used for boolean checks, while `assertEqual` should be used to compare values.